[#5943] Email subscription on /download/thanks/

craigcook opened the pull request:
## Description
Adds a newsletter signup form on the /download/thanks/ page. 

This is a Traffic Cop A/B experiment with two headline variations, with a 50/50 split for all en-US users. The experiment uses the switch `firefox_download_scene2_email_experiment`

Targeting English-only for now so no l10n needed. If this test proves successful I imagine a version of this would become the new default /thanks/ page for all locales but would likely be redesigned by then.

Also adds a new email privacy page at /privacy/email/. The link to that page may become part of the standard form and we'll need to localize the page at that point.

## Issue / Bugzilla link

## Testing
[Version A](https://bedrock-demo-craigcook.oregon-b.moz.works/en-US/firefox/download/thanks/?v=a)
[Version B](https://bedrock-demo-craigcook.oregon-b.moz.works/en-US/firefox/download/thanks/?v=b)
[Email privacy](https://bedrock-demo-craigcook.oregon-b.moz.works/en-US/privacy/email/)

craigcook commented:
@agibson This will surely cause some merge conflicts with https://github.com/mozilla/bedrock/pull/5999 though this one probably needs to merge first since it includes the /privacy/email page... hopefully the rebase won't be too hairy.

alexgibson commented on media/js/firefox/new/scene2-email.js:
```diff
@@ -0,0 +1,52 @@
+(function($) {
+    'use strict';
+
+    var isIELT9 = window.Mozilla.Client.platform === 'windows' && /MSIE\s[1-8]\./.test(navigator.userAgent);
```
All this JS looks the same as the regular `scene2.js` code. Rather than have to maintain all this JS twice, I'd suggest putting the new modal-related JS code into a separate file that we can include in the template, in addition to the regular `scene2.js` file?

craigcook replied:
Good call. It is indeed just a copy of scene2.js with the modal added, no reason for the repetition. I'll update.

alexgibson commented on bedrock/firefox/views.py:
```diff
@@ -573,6 +573,10 @@ def download_thanks(request):
             template = 'firefox/new/portland/scene2-safe.html'
         elif experience == 'betterbrowser':
             template = 'firefox/new/better-browser/scene2.html'
+        elif switch('firefox_download_scene2_email_experiment') and variant == 'a':
```
Maybe I'm a little confused on the plan, but I thought we wanted to roll this new page out as the default because it was the "right thing to do"? I had imagined the A/B copy test would be separate to doing that - but I'm guessing I didn't understand quite right.

craigcook replied:
I'll get clarification on the test. I was under the impression this was going to be temporary and would need to be switch-offable, though I don't know what we're measuring against so maybe it really is the new permanent default (for en-US).

alexgibson commented:
The design has changed since I last saw it, but is something supposed to be over to the right of the page? At first i thought I was missing an image file locally, but it seems to be the same on demo. Visually it feels like something should be there, or if not the newsletter could be centered on the page.

![screenshot_2018-08-14 download the fastest firefox ever](https://user-images.githubusercontent.com/400117/44088496-313d82ba-9fbb-11e8-933a-820c11b0df36.png)

alexgibson commented on media/js/firefox/new/experiment-download-scene2-email.js:
```diff
@@ -0,0 +1,17 @@
+    var cop = new Mozilla.TrafficCop({
+        id: 'experiment-download-scene2-email-082018',
+        variations: {
+            'v=a': 50,
```
Which is the control group here? The new page isn't made default, but we're still putting 100% of users into the copy A/B test. Usually here we'd add a control group to compare conversions against, which normally defaults to the current experience in prod. It feels like that's not what we're testing here though, we're just testing the copy. Is there any reason why we can't make this new page default regardless of the experiment? Then we could treat say, version a, as the control?

alexgibson commented on bedrock/firefox/views.py:
```diff
@@ -573,6 +573,10 @@ def download_thanks(request):
             template = 'firefox/new/better-browser/scene2.html'
+        elif switch('firefox_download_scene2_email_experiment') and variant == 'a':
+            template = 'firefox/new/email/scene2-a.html'
+        elif switch('firefox_download_scene2_email_experiment') and variant == 'b':
```
A couple of Python unit test here would be good?

alexgibson commented on media/css/firefox/new/scene2_email.scss:
```diff
@@ -0,0 +1,240 @@
+
+
+
+
```
Nit blank lines

alexgibson commented on bedrock/firefox/templates/firefox/new/email/scene2-base.html:
```diff
@@ -0,0 +1,114 @@
+      <p class="tagline">{{ _('Get informative tips, tricks and product announcements delivered to your inbox.') }}</p>
+
+      <div class="c-newsletter">
+        {{ email_newsletter_form(
```
Not something we necessarily need to fix right now, but what do people see if they don't get the newsletter translated?

craigcook replied:
Good question... since this is English only for now it's not an issue but I don't know what the longer term plan is. There's talk of soon redesigning the default /new as well as /thanks so things are likely to change. When we have a new default thanks page for all locales we'll definitely need a non-newsletter fallback. I'd suggest Fx mobile, like on the current page.

craigcook commented:
OK, made various updates...

* Switched to Protocol
* Simplified JS
* Made this the default page for English, no switch, with variant A as the control
* Added and updated tests

djst commented:
I don't know if you sorted this already, but we should roll out the test
50%/50% split. It is indeed "the right thing to do" so there's no need for
a control outside of these two variants. We are just trying to see which of
these two performs the best. As soon as we have a winner, we'll end the
test and roll out the winner to 100%. The previous thank you page without a
newsletter signup tout is not coming back. Hope that clarifies.

Den ons 15 aug. 2018 kl 00:53 skrev Craig Cook <notifications@github.com>:

> OK, made various updates...
>
>    - Switched to Protocol
>    - Simplified JS
>    - Made this the default page for English, no switch, with variant A as
>    the control
>    - Added and updated tests
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/bedrock/pull/6008#issuecomment-413041419>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFYHom0Zndt-11X89N_XNEueUuwzJsSrks5uQ1TlgaJpZM4V7jur>
> .
>

alexgibson commented on bedrock/firefox/templates/firefox/new/email/scene2.html:
```diff
@@ -11,6 +17,11 @@
   <link rel="canonical" href="{{ settings.CANONICAL_URL }}/{{ LANG }}/firefox/new/">
 {% endblock %}
 
+{% block page_image %}{{ static('img/firefox/template/page-image.png') }}{% endblock %}
```
Rather than add all these pieces of metadata on a per-page basis, would it make sense to create a `firefox/base-protocol.html` file that this template extends? (similar to how we already do for pebbles). That way all new Protocol pages under `/firefox/` could get this meta data for free?

alexgibson commented on media/css/firefox/new/scene2_email.scss:
```diff
@@ -120,103 +121,43 @@
-        padding-left: 2em;
-        display: block;
+            background: $color-white;
+            box-shadow: 4px 4px 0 0 #000;
```
Buttons are always a little contentious, but we seem to be making quite a few button related tweaks on Protocol pages that have not yet been adopted in the design system. Being forward thinking is good, but the more we do this the more papercuts we create for ourselves in the future to go back and undo all these custom styles. I guess this is up to our designers, but imho the UX here is not going to be negatively impacted if we just use the stock button for now :/

craigcook replied:
I'm working on a PR to make this the base button style in Protocol. But yeah, this page would probably be fine with the current Protocol button until it's updated in Protocol (at which point this would automatically update once the new button is published upstream; the system works!) I'll remove this extra styling.

craigcook replied:
Oh now I remember... the default button switches to white in a dark theme but Jennifer wanted it to be black here so it would be distinct from the form fields. So since I needed to invert the color anyway that's why I added the shadow... so it seems this still needs to be a non-standard button for the moment, but I'll at least remove the shadow and just invert the colors. 

And I'll make sure to include a theme class when I update the buttons in Protocol so we can still have a black button in a dark theme when needed.

alexgibson commented on media/js/firefox/new/scene2-email.js:
```diff
@@ -2,51 +2,43 @@
-    var $modalLink = $('.email-privacy-link');
-
-    $modalLink.on('click', function(e) {
+    // var $modalWrapper = $('.email-privacy');
```
Can this commented out code be removed now?

craigcook replied:
doh! I'll clean up my mess.

alexgibson commented on media/static-bundles.json:
```diff
@@ -979,7 +978,10 @@
-        "js/base/mozilla-modal.js",
+        "protocol/js/protocol-modal.js",
+        "protocol/js/protocol-newsletter.js",
+        "js/newsletter/form-protocol.js",
```
Ugh, I totally forgot we still need to move that code into Protocol 😢

alexgibson commented on media/js/firefox/new/scene2-email.js:
```diff
@@ -2,51 +2,43 @@
+        Mozilla.Modal.createModal(e.target, content, {
+            title: title.innerHTML,
+            className: 'mzp-t-firefox',
+            closeText: strings.dataset.close
```
Nit: use `Mozilla.Utils.trans()` here for browsers that don't support native `dataset`.

alexgibson commented:
> I don't know if you sorted this already, but we should roll out the test 50%/50% split. It is indeed "the right thing to do" so there's no need for a control outside of these two variants. We are just trying to see which of these two performs the best. As soon as we have a winner, we'll end the test and roll out the winner to 100%. The previous thank you page without a newsletter signup tout is not coming back. Hope that clarifies.

Right - but all tests should have a control group. If all we're testing is the copy, then one of the copy variations should be the control. So what we're doing is this:

- Make variation a of this new page the default for en-US .
- 50% of visitors to this page may see a different copy (variation b).
- Conversions are compared against the other 50% (variation a, control).

Hope that makes sense.

alexgibson commented on bedrock/firefox/templates/firefox/new/email/scene2.html:
```diff
@@ -0,0 +1,132 @@
+ # License, v. 2.0. If a copy of the MPL was not distributed with this
+ # file, You can obtain one at http://mozilla.org/MPL/2.0/. -#}
+
+{% extends "base-protocol.html" %}
```
If I set `Dev=False`, I don't see the language switcher in the footer for this page, so i think we need to add `{% add_lang_files "firefox/new/quantum" %}` here?

alexgibson commented:
If I visit the `/thanks/` page in an old legacy IE, everything looks presentable enough, but the newsletter form won't expand, so I can't submit an email sucessfully. The form should work without JS still, so maybe we can add a CSS override to in oldIE.scss to make sure it's always expanded?

<img width="594" alt="screen shot 2018-08-15 at 12 27 00" src="https://user-images.githubusercontent.com/400117/44146216-929f1f52-a086-11e8-8336-418d11c0056c.png">

djst commented:
Yep, makes total sense.

Den ons 15 aug. 2018 kl 11:53 skrev Alex Gibson <notifications@github.com>:

> I don't know if you sorted this already, but we should roll out the test
> 50%/50% split. It is indeed "the right thing to do" so there's no need for
> a control outside of these two variants. We are just trying to see which of
> these two performs the best. As soon as we have a winner, we'll end the
> test and roll out the winner to 100%. The previous thank you page without a
> newsletter signup tout is not coming back. Hope that clarifies.
>
> Right - but all tests should have a control group. If all we're testing is
> the copy, then one of the copy variations should be the control. So what
> we're doing is this:
>
>    - Make variation a of this new page the default for en-US .
>    - 50% of visitors to this page may see a different copy (variation b).
>    - Conversions are compared against the other 50% (variation a,
>    control).
>
> Hope that makes sense.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/bedrock/pull/6008#issuecomment-413149667>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFYHogIYMCzM4ZnJt7GVeqD4dCzVYLBIks5uQ--fgaJpZM4V7jur>
> .
>

craigcook commented:
> If I visit the /thanks/ page in an old legacy IE, everything looks presentable enough, but the newsletter form won't expand, so I can't submit an email sucessfully. The form should work without JS still, so maybe we can add a CSS override to in oldIE.scss to make sure it's always expanded?

Which version of IE? Did we ever fully drop all JS from IE8 and older? The form is only hidden when JS is available so evidently we're still adding the `.js` class in old IEs in order to hide things but not serving them the JS that makes them visible again.

alexgibson commented:
> Which version of IE? Did we ever fully drop all JS from IE8 and older? The form is only hidden when JS is available so evidently we're still adding the .js class in old IEs in order to hide things but not serving them the JS that makes them visible again.

We don't serve first class CSS or JS to < IE9.

craigcook commented:
OK, *all* browsers still get `site.js` which is what adds the `js` class to root, but lte IE8 don't get the newsletter JS. I think rather than trying to so any fancy JS hoops a simple override in the CSS is better. Here comes `!important`...

craigcook commented:
Hold on, waiting for a small copy change on the email privacy page, as well as final signoff from Michelle. Marking DNM for the moment.
